streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -672,6 +678,8 @@ private int restoreChangelog(final Task task, final ChangelogMetadata changelogM
                 } catch (final Exception e) {
                     throw new StreamsException("State restore listener failed on batch restored", e);
                 }
+            } else if (changelogMetadata.stateManager.taskType() == TaskType.STANDBY) {
+                standbyUpdateListener.onBatchLoaded(partition, storeName, stateManager.taskId(), currentOffset, numRecords, storeMetadata.endOffset());
```

Comment:
We should probably put this in a try block like the restore listener does right above this:
```suggestion
                try {
                    standbyUpdateListener.onBatchLoaded(partition, storeName, stateManager.taskId(), currentOffset, numRecords, storeMetadata.endOffset());
                } catch (final Exception e) {
                    throw new StreamsException("Standby updater listener failed on batch loaded", e);
                }
```